src/components/Article.jsx

```diff
@@ -29,6 +30,32 @@ const Article = () => {
             })
     }, [params.article_id])
 
+    const downvote = () => {
+        setUserVotes(currentUserVotes => {
+            return currentUserVotes - 1
+        })
+        patchArticleById(params.article_id, -1)
+            .catch(err => {
+                setUserVotes(currentUserVotes => {
+                    return currentUserVotes + 1
+                })
+                setIsVoteError(true)
+            })
+    }
+
+    const upvote = () => {
+        setUserVotes(currentUserVotes => {
+            return currentUserVotes + 1
+        })
+        patchArticleById(params.article_id, 1)
+            .catch(err => {
+                setUserVotes(currentUserVotes => {
+                    return currentUserVotes - 1
+                })
+                setIsVoteError(true)
+            })
+    }
```

Comment:
Could you make a dynamic function that could do both upvote and downvote? It would make this drier :)